src/processors/ScrsProcessor.ts

```diff
@@ -0,0 +1,123 @@
+                }
+
+                const scResults: SmartContractResult[] = transactions.get(data.scrs[key].originalTxHash);
+                scResults.push(data.scrs[key]);
+
+                transactions.set(data.scrs[key].originalTxHash, scResults);
+            }
+        }
+
+        await this._processListTransactions(transactions);
+    }
+
+    private async _processListTransactions(transactions: Map<string, SmartContractResult[]>): Promise<void> {
+        const promises: Promise<ListNftApiResponse>[] = [];
+
+        for (const [originalTxHash, scResults] of transactions) {
+            const { listOpArgs, opResultArgs, ownerAddress } = this._getListTransactionDetails(scResults);
+
+            if (!listOpArgs || !opResultArgs) {
+                // TODO: should create an analytic class responsible for recording how many transactions failed
+                console.warn(`It is not a list transaction: ${originalTxHash} -> ${JSON.stringify(scResults)}: `, { listOpArgs, opResultArgs, ownerAddress });
+                continue;
+            }
+
+            let listArgsBuilder: ListArgsBuilder = new ListArgsBuilder();
+            let tokens: string[] = listOpArgs.split("@");
+
+            if (tokens.length === 5) {
+                /* add collection ticker and nonce */
+                listArgsBuilder = listArgsBuilder.addCollectionTicker(Utility.convertHexToString(tokens[1])).addNonce(tokens[2]);
+
+                /* add price */
+                listArgsBuilder = listArgsBuilder.addPrice(Utility.convertHexToDecimal(tokens[3]) / 10000000000 / 100000000);
+
+                /* add custom royalties */
+                listArgsBuilder = listArgsBuilder.addCustomRoyalties(Utility.convertHexToDecimal(tokens[4]) / 100);
+
+                /* add owner address */
+                listArgsBuilder = listArgsBuilder.addOwnerAddress(Utility.convertBase64ToBech32Address(ownerAddress));
+                tokens = opResultArgs.split("@");
+
+                if (tokens.length === 3) {
+                    /* add transaction id and build the final object */
+                    const listArgs: ListArgs = listArgsBuilder.addTransactionId(tokens[2]).build();
+                    const url: string = `${Config.NFTR_HOST}${Config.LIST_NFT_URI}`;
+
+                    promises.push(this._listNftApiCaller.call(url, { body: { ...listArgs } }));
+                } else {
+                    console.warn(`Not enough arguments for list op result. Expected 3, got ${tokens}`);
+                }
+            } else {
+                console.warn(`Not enough arguments for list op. Expected 5, got ${tokens}`);
+            }
+            
+        }
+
+        const results: PromiseSettledResult<ListNftApiResponse>[] = await Promise.allSettled(promises);
+
+        results.forEach((result: PromiseSettledResult<ListNftApiResponse>) => {
+            if (result.status === "rejected") {
```

Comment:
@trascaandrei  what about save to a queue these rejected promises and try later again?
Why could these promises be rejected? Maybe the nftr BE is down or a network problem occurred.
I am not sure what will be the impact if a list transaction is not store into nftr DB.

Reply:
Yes, definitely. The impact is the fact that those NFT won't be available on the website and it is exactly the problem that we are trying to solve with this observer

Reply:
I will add a task on Jira for implementing a retry mechanism for these rejected requests.

Reply:
Jira task: https://trsdesignro.atlassian.net/browse/NFTR-29